src/libstd/sys/common/wtf8.rs

```diff
@@ -426,26 +426,35 @@ impl Ord for Wtf8 {
 /// and surrogates as `\u` followed by four hexadecimal digits.
 /// Example: `"a\u{D800}"` for a slice with code points [U+0061, U+D800]
 impl fmt::Debug for Wtf8 {
-    fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        fn write_str_escaped(f: &mut fmt::Formatter, s: &str) -> fmt::Result {
+            for c in s.chars().flat_map(|c| c.escape_default()) {
+                try!(write!(f, "{}", c))
```

Comment:
I think we want this to just be `try!(f.write_char(c))`, it saves some indirections.